---
title: "5.4.8-9. Differential gene expression and gene set analysis."
output: 
  revealjs::revealjs_presentation:
    theme: night
# knit: my_render
# theme specifies the theme to use for the presentation (available themes are "default", "simple", "sky", "beige", "serif", "solarized", "blood", "moon", "night", "black", "league" or "white").
# © Abel Vertesy 2018
---




## 5.4.8. Finding differentially expressed genes (DEG)

>- Common analysis when working with bulk RNA-seq
>   - 2 conditions, e.g. mutant and wild-type
> -  Common packages:
>     - [DESeq2](https://bioconductor.org/packages/DESeq2) and
>     - [edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html),
> -  But single-cell data is somewhat different, so

​    

## Differential gene expression in single-cell data

Differences

- Usually do not have a defined
  set of experimental conditions
  - Rather you derive the clusters
    - Cluster vs background (all)
    - Cluster 1 vs Cluster 2
- The distribution of data is very different: think of the zeros.
  - →  Assumptions of bulk methods might not be valid
  -  → New methods are designed for `scDE`, and they are typically part of pipelines.


## 5.4.9. Gene set analysis

Problem:

- Many DEG, but what do they really mean?
- Is there a common function?
- GO-term analysis can help summarising the results.



## Each gene is associated with GO-terms

- Calculate *enrichment*.

$$Terms.in.your.genes \over Terms.in.ALL.genes$$



## GO-term annotation & protein interaction networks

- Open up STRING website for interactive exploration



